internal/transport/http2_client.go

```diff
@@ -367,6 +371,9 @@ func (t *http2Client) newStream(ctx context.Context, callHdr *CallHdr) *Stream {
 			closeStream: func(err error) {
 				t.CloseStream(s, err)
 			},
+			freeBuffer: func(b *bytes.Buffer) {
```

Comment:
can you assign `freeBuffer` to `t.bufferPool.put` directly and avoid the lambda?  (I assume `t` and `t.bufferPool` do not change over time?)